src/charm.py

```diff
@@ -197,47 +214,51 @@ def _get_version(self) -> Optional[str]:
     def _set_version(self) -> None:
         if version := self._get_version():
             self.unit.set_workload_version(version)
-
-    # Properties
+            logger.info(f"Set workload version: {version}")
 
     @property
     def _admin_ui_pebble_layer(self) -> Layer:
-        """Define container configuration."""
-        container = {
-            "override": "replace",
-            "summary": "identity platform admin ui",
-            "command": ADMIN_UI_COMMAND,
-            "startup": "enabled",
-            "environment": {
-                "KRATOS_PUBLIC_URL": self._kratos_public_url,
-                "KRATOS_ADMIN_URL": self._kratos_admin_url,
-                "HYDRA_ADMIN_URL": self._hydra_admin_url,
-                "IDP_CONFIGMAP_NAME": self._idp_configmap_name,
-                "IDP_CONFIGMAP_NAMESPACE": self._idp_configmap_namespace,
-                "SCHEMAS_CONFIGMAP_NAME": self._schemas_configmap_name,
-                "SCHEMAS_CONFIGMAP_NAMESPACE": self._schemas_configmap_namespace,
-                "PORT": self._port,
-                "TRACING_ENABLED": False,
-                "LOG_LEVEL": self._log_level,
-                "LOG_FILE": str(LOG_FILE),
-                "DEBUG": self._log_level == "DEBUG",
-            },
+        """Define pebble layer."""
+        container_env = {
+            "KRATOS_ADMIN_URL": self._get_kratos_endpoint_info("admin_endpoint"),
+            "KRATOS_PUBLIC_URL": self._get_kratos_endpoint_info("public_endpoint"),
+            "HYDRA_ADMIN_URL": self._get_hydra_endpoint_info(),
+            "IDP_CONFIGMAP_NAME": "providers",
+            "IDP_CONFIGMAP_NAMESPACE": self.model.name,
```

Comment:
AFAICT If kratos is deployed in a different model this will fail, not sure if we care about this but I think we should document this for future reference (a simple comment would be enough)